eng/run.ps1

```diff
@@ -99,12 +99,18 @@ try {
   # Use a module-local path so Go resolves imports correctly.
   $module_local_script_path = Join-Path "." "cmd" "$tool"
 
-  Write-Host "In '$tool_module', building '$module_local_script_path' -> $tool_output"
-  & (Join-Path $stage0_goroot "bin" "go") build -o $tool_output $module_local_script_path
-  if ($LASTEXITCODE) {
-    Write-Host "Failed to build tool."
-    exit 1
+  # The caller may have passed in GOOS/GOARCH to cross-compile Go. We can't use those values here:
+  # we need to be able to run the tool on the host, so we must always target the host OS/ARCH. Clear
+  # out the GOOS/GOARCH values (empty string) to detect host OS/ARCH automatically for the tools.
+  Invoke-CrossGoBlock "" "" {
+    Write-Host "In '$tool_module', building '$module_local_script_path' -> $tool_output"
```

Comment:
One thing to be careful about is how block execution and variable scoping works in powershell. I've gotten burned by this in the past because I made bad assumptions about how name lookup works here. IIRC the variables are not actually captured, aka it's not a lambda, but instead the variable lookup occurs in the context where the block is evaluated. 

Been many years since I got burned here so maybe I'm having bad memory but I would look into this a bit.

Reply:
That's right, no capturing.

In this particular case, capture vs. non-capture doesn't have any impact because `Invoke-CrossGoBlock` doesn't set any variables that are being used in the block, the block is defined right here where it's being passed to the function, and the block doesn't escape. But definitely something to be aware of while modifying these scripts.

I could inline `Invoke-CrossGoBlock` to avoid the weirdness, but we rely on blocks for `Invoke-WithRetry` too, so it wouldn't really remove the need to know about this quirk.

I'm glad we're able to get out of this script and into the _core Go module relatively quickly. 😄